query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -66,25 +66,59 @@ trait MetadataDistConcatExec extends NonLeafExecPlan {
   }
 }
 
+case object SerializeUtils {
+  def serialize(obj: Any) : Array[Byte] = {
+    val byte_os = new ByteArrayOutputStream()
+    val obj_os = new ObjectOutputStream(byte_os)
+    obj_os.writeObject(obj)
+    val res = byte_os.toByteArray
+    byte_os.close()
+    obj_os.close()
+    res
+  }
+  def deSerialize[T](bytes: Array[Byte]): T = {
+    val byte_is = new ByteArrayInputStream(bytes)
+    val obj_is = new ObjectInputStream(byte_is)
+    val res = obj_is.readObject().asInstanceOf[T]
+    byte_is.close()
+    obj_is.close()
+    res
+  }
+}
+
```

Comment:
Not 100% sure this is the best way to go about things, but it seems to work

Reply:
I see now that this actually encodes the fully-qualified names of every object-- working on a better fix, now

Reply:
I realize now that this encodes a fully-qualified type with every single reference. Fixing now...

Reply:
Fixed in recent commit